govcd/extension.go

```diff
@@ -47,6 +49,39 @@ func GetExternalNetworkByName(vcdClient *VCDClient, networkName string) (*types.
 	return &types.ExternalNetworkReference{}, nil
 }
 
+func CreateExternalNetwork(vcdClient *VCDClient, externalNetwork *types.ExternalNetwork) (Task, error) {
+	err := validateExternalNetwork(externalNetwork)
+	if err != nil {
+		return Task{}, err
+	}
+
+	output, err := xml.MarshalIndent(externalNetwork, "  ", "    ")
+	if err != nil {
+		return Task{}, fmt.Errorf("error marshalling xml: %s", err)
+	}
+	xmlData := bytes.NewBufferString(xml.Header + string(output))
+	util.Logger.Printf("[TRACE] CreateExternalNetwork - xml payload: %s\n", xmlData)
+	externalnetsHREF := vcdClient.Client.VCDHREF
+	externalnetsHREF.Path += "/admin/extension/externalnets"
+	req := vcdClient.Client.NewRequest(map[string]string{}, "POST", externalnetsHREF, xmlData)
+	req.Header.Add("Content-Type", "application/vnd.vmware.admin.vmwexternalnet+xml")
+	resp, err := checkResp(vcdClient.Client.Http.Do(req))
+	if err != nil {
+		util.Logger.Printf("[TRACE] error instantiating a new ExternalNetwork: %s", err)
+		return Task{}, fmt.Errorf("error instantiating a new ExternalNetwork: %s", err)
+	}
+
+	externalNetwork = new(types.ExternalNetwork)
+	if err = decodeBody(resp, externalNetwork); err != nil {
+		util.Logger.Printf("[TRACE] error decoding admin extension externalnets response: %s", err)
+		return Task{}, fmt.Errorf("error decoding admin extension externalnets response: %s", err)
+	}
+
+	task := NewTask(&vcdClient.Client)
+	task.Task = externalNetwork.Tasks.Task[0]
```

Comment:
Doesn't work directly parsing the task as we do in other places? for e.g.
```
	task := NewTask(eGW.client)

	if err = decodeBody(resp, task.Task); err != nil {
		return Task{}, fmt.Errorf("error decoding Task response: %s", err)
	}
```

Reply:
It does!

Reply:
So I think it can be improved by decoding `Task` directly and avoid the not needed checks and getting from array.

Reply:
Agreed! I have already made that change in one of the newer commits.